public/app/features/dashboard-scene/scene/DashboardControls.tsx

```diff
@@ -151,11 +153,63 @@ function DashboardControlsRenderer({ model }: SceneComponentProps<DashboardContr
   const showDebugger = window.location.search.includes('scene-debugger');
   const hasDashboardControls = useHasDashboardControls(dashboard);
 
+  // Get adhoc and groupby variables for drilldown controls
+  const { variables } = sceneGraph.getVariables(dashboard)?.useState() ?? { variables: [] };
+  const visibleVariables = variables.filter((v) => v.state.hide !== VariableHide.inControlsMenu);
+  const adHocVar = visibleVariables.find((v) => sceneUtils.isAdHocVariable(v));
+  const groupByVar = visibleVariables.find((v) => sceneUtils.isGroupByVariable(v));
```

Comment:
It's probably a very niche scenario, but this fill find only first ad hoc and group by, while many can be configured. I don't think this is to be addresed in this PR. however it's something we need to keep in mind in the future if we decide to merge filters and group bys under a unified UI @Ijin08